### PR TITLE
updates to chapter 5.10

### PR DESCRIPTION
**Pull request by ritikarawlani**
Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed
- [ ] I have selected a committee co-chair to review the PR

### CODE REVIEW

File: ch-5.10.html [hunk 1]
```diff
@@ -81,28 +81,28 @@ <h4 id="5.10.1.2">5.10.1.2 Informative References</h4>
...
-        </li>
-        <li class="bullet-list1">Shall conform to JAdES-B-LT – for support of Long Term signatures. The JAdES-B-LT specification adds the timestamp of the signing, inclusion of the signing certificate and statement of revocation.</li>
+        <li class="bullet-list1"><p><strong>SHALL</strong> use JSON Web Signature (JWS)(see <a href="https://tools.ietf.org/html/rfc7515">RFC 7515</a>). JSON Web Signature (JWS) is a means of representing content secured with digital signatures or Hash-based Message Authentication Codes (HMACs) using JSON data structures. Cryptographic algorithms and identifiers used with this specification are enumerated in the separate JSON Web Algorithms (JWA)</li>
+        <li class="bullet-list1">JWS JSON serialization shall be used as specified in <a href="https://tools.ietf.org/html/rfc7515">RFC 7515</a></li>
```
Comment (ritikarawlani) [inline ch-5.10.html:85]: make it specific to JWS so as to not confuse with payload

File: ch-5.10.html [hunk 2]
```diff
@@ -118,61 +118,65 @@ <h5 id="5.10.2.1.6">5.10.2.1.6 "sigPId" (signature Policy Identifier) Header Par
...
+      <h3 id="5.10.3">5.10.3 JSON Detached Signature</h4>
+      <h4 id="5.10.3.1">5.10.3.1 Protected Header</h4>
         <h5 id="5.10.3.1.1">5.10.3.1.1 "sigD" Header Parameter</h5>
-        <ul><li>sigD parameter shall be included as per 5.2.8.1 of the JADES Specification</li>
+        <ul>
+          <li>sigD parameter shall be included as per 5.2.8.1 of the JAdES Specification</li>
             <li>mID member shall be present and set to "http://uri.etsi.org/19182/ObjectIdByURIHash"</li>
+            
+              <li class="bullet-list1"><mark>
+                pars member shall shall be used and hold the document uniqueID. For documents that do not use a URI as the uniqueId, the Affinity Domain should determine an appropriate way to encode the DocumentEntry.uniqueId</mark>
```
Comment (ritikarawlani) [inline ch-5.10.html:141]: figure out the tagging (em?)

File: ch-5.10.html [hunk 3]
```diff
@@ -118,61 +118,65 @@ <h5 id="5.10.2.1.6">5.10.2.1.6 "sigPId" (signature Policy Identifier) Header Par
...
+      <p>No additional considerations other than the ones specified in the section <a href="5.10.2.2">5.10.2.2</a></p>
+      <h4 id="5.10.4.3">5.10.4.3 Payload</h4>
+       <ul><li> When FHIR Resources are signed, the signature is across the BASE64URL(JWS Payload) form of the Payload. The DSGj profile does not create any additional expectations of transformation on the Payload of any form.</ul></li>
+      <h4 id="5.10.4.4">5.10.4.4 Signature</h4>
+      <p>No additional considerations other than the ones specified in the section <a href="5.10.2.4">5.10.2.4</a></p>
       <h3 id="5.10.5">5.10.5 JSON Signature Verification</h3>
       <p>There are three levels of signature verification:</p>
       <ol>
-        <li class="numbered-list1">verifying that the Digital Signature block itself has integrity through verifying the signature across the XML-Signature,</li>
+        <li class="numbered-list1">verifying that the JWS JSON object itself has integrity through verifying the signature across the Signature,</li>
```
Comment (ritikarawlani) [inline ch-5.10.html:169]: remove 'across the Signature'

File: ch-5.10.html [hunk 4]
```diff
@@ -200,20 +204,21 @@ <h5 id="5.10.6.1.3">5.10.6.1.3 XDSDocumentEntry.typeCode</h5>
...
       <h5 id="5.10.6.1.7">5.10.6.1.7 XDSDocumentEntry.title</h5>
       <p>Should be the same as the display name for the signature purpose</p>
       <h5 id="5.10.6.1.8">5.10.6.1.8 XDSDocumentEntry.language</h5>
       <p>The language of the signature content shall be ‘art’ as in “artificial”.</p>
-      <h4 id="5.10.6.2">5.10.6.2 Document Sharing – SubmissionSet Metadata</h4>
-      <p>This document content profile makes no changes to the structure of Submission Sets.</p>
+   <mark>
+        <h4 id="5.10.6.2"><mark>5.10.6.2 Document Sharing – SubmissionSet Metadata</mark></h4>
+        <p>This document content profile makes no changes to the structure of Submission Sets.</p>
+   </mark>
```
Comment (ritikarawlani) [inline ch-5.10.html:217]: remove it

**Review by JohnMoehrke — COMMENTED** [review]
For the listing of documents signed. where we need URI encoding of the document id. There is already a well-defined URI encoding for most of the document uniqueId values
in the definition of uniqueId.
https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.26

except that definition forbids using the urn:hl7ii. This was forbidden due to precedence of not using it historically. This was not forbidden because of some technical reason. so in the context of DSG we could just require it, since we must have URI.

**Comment by JohnMoehrke** [comment]
> For the listing of documents signed. where we need URI encoding of the document id. There is already a well-defined URI encoding for most of the document uniqueId values in the definition of uniqueId. https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.26
> 
> except that definition forbids using the urn:hl7ii. This was forbidden due to precedence of not using it historically. This was not forbidden because of some technical reason. so in the context of DSG we could just require it, since we must have URI.

Discussion with @jlamy ; IHE has not yet defined a URI encoding for homeCommunityID and uniqueID... Adding the homeCommunityID is not required, it is more of a nice-to-have, as the uniqueID is globally unique. But adding the homeCommunityId does enable discovery if one doesn't already know where that document exists. Not likely in a signature validation, but possible that the signature you need to validate for one document covered two documents so you need to go get the other document simply to validate the signature over the document you have. Thus knowing which homeCommunityID to search within would be nice. You can, in theory, search the whole connected federation of HIE for your document uniqueId, but that is expensive and might be not allowed. SO, we could say that if you want to indicate the homeCommunityID; one appends it to the uniqueId (URL) with a '#' or '|' separator (we would need to pick one).

**Comment by litlfred** [comment]
If we were to support this, I would prefer that we didn't make use of sting
concatenation as it SHALL* make it harder downstream to extract those
values.   I think there may be flexibility to add an (optional) protected
header to capture the homeCommunityID.

(Apologies for my lack of familiarity with XDS) My understanding is that
homeCommunityID should be the ID of the regional network/community which is
comprised of multiple participants/jurisdictions that issue/own health
documents.

Can a jurisdiction which participates in multiple network/communities and
issue the same document for multiple communities/networks at once?  If so,
would we need multiple homeCommunityID's?



Cheers,
-carl

* :-)

**Comment by JohnMoehrke** [comment]
Each document may exist in a different community. The most likely is that the community is not important. Most likely the signature is on one document that is in one community. We could possibly not address it, or at least move it to an open-issue. 

I have not yet exhausted research to find an existing mechanism for indicating a document uniqueID within a community.

**Comment by ritikarawlani** [comment]
@JohnMoehrke  Action item to add serialization note for documents in Volume 1, and refer to PCC1

**Comment by ritikarawlani** [comment]
@JohnMoehrke shall I merge this PR ahead of creating a new commit tomorrow?

**Comment by JohnMoehrke** [comment]
yes, I think we should merge the PR once we review it. That way we can
focus on the new updates.

John Moehrke 🔥 Architect: Healthcare Informatics Standards -
Interoperability, Privacy, and Security
IHE Co-Chair IT Infrastructure Planning and Technical
HL7 Co-Chair Security WG, FHIR FMG, FHIR facilitator, and FHIR Foundation
founding member
Employee of By Light -- Contractor to VHA MyHealtheVet
***@***.***  |  M +1 920-564-2067  |  ***@***.***
 https://healthcaresecprivacy.blogspot.com

**Pull request merged**